services/appflowy-collaborate/src/collab/queue.rs

```diff
@@ -468,12 +469,21 @@ async fn write_pending_to_disk(
   }
 
   // Commit the transaction to finalize all writes
-  transaction
-    .commit()
-    .await
-    .context("Failed to commit the transaction for pending collaboration data")
-    .map_err(AppError::from)?;
-  Ok(success_write_objects)
+  match tokio::time::timeout(Duration::from_secs(10), transaction.commit()).await {
+    Ok(result) => {
+      result.map_err(AppError::from)?;
+      Ok(success_write_objects)
+    },
+    Err(_) => {
```

Comment:
Is timeout the only reason for commit failure? Should we include the error message in the log as well?